### PR TITLE
Add packing envelopes

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/cgal.cc [hunk 1]
```diff
@@ -4779,6 +4695,105 @@ void FromSurfaceMeshToPolygonsWithHoles(const Surface_mesh* input,
...
+      PS::simplify(ct, cid, cost, stop, true);
+    }
+
+    // And emit.
+    for (CT::Constraint_id cid : cids) {
+      emit_polygon(false);
+      Vertices_in_constraint_iterator it = ct.vertices_in_constraint_begin(cid);
+      Vertices_in_constraint_iterator end = ct.vertices_in_constraint_end(cid);
+      if (it != end) {
+        for (;;) {
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal.cc:4766]: I had to look up how this works, it's neat. I didn't realize that ALL of the arguments to for() are optional

File: algorithm/cgal/insetOfPolygonWithHoles.test.js [hunk 2]
```diff
@@ -16,128 +16,136 @@ test('insetOfPolygonWithHoles/bow', (t) => {
...
+        [0.109375, 0.048564189189189186, 0],
+        [-0.0894736842105263, 0.1631282959789057, 0],
       ],
       exactPoints: [
-        ['-17/190', '-826757/5324670', '0'],
-        ['9/52', '-4/999', '0'],
-        ['373/1924', '0', '0'],
-        ['9/52', '4/999', '0'],
-        ['-17/190', '826757/5324670', '0'],
+        ['-17/190', '-79189/485440', '0'],
```
Comment (BarbourSmith) [inline algorithm/cgal/insetOfPolygonWithHoles.test.js:90]: What was the change to inset? Was it the addition of segments?...or was that already there?

File: nb/projects/plotter/plotter.stars_0.gcode [hunk 3]
```diff
@@ -2,377 +2,270 @@
 G21
```
Comment (BarbourSmith) [inline nb/projects/plotter/plotter.stars_0.gcode]: We might want to add a G90 or G91 to the front of these to indicate absolute vs incremental moves. G90 is absolute.

**Review by BarbourSmith — APPROVED** [review]
Looks great!

File: algorithm/cgal/cgal.cc (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/cgal.cc:4766]: It's not that they're optional, but rather that an empty statement is valid. :)

File: algorithm/cgal/insetOfPolygonWithHoles.test.js (hunk 2 above)
Comment (pentacular) [reply algorithm/cgal/insetOfPolygonWithHoles.test.js:90]: The additional of parameterized segments, I think.

File: nb/projects/plotter/plotter.stars_0.gcode (hunk 3 above)
Comment (pentacular) [reply nb/projects/plotter/plotter.stars_0.gcode]: Ah, good point.

**Pull request merged**